src/util/irep_hash_container.h

```diff
@@ -55,9 +57,10 @@ class irep_hash_container_baset
   {
     inline size_t operator()(const packedt &p) const
     {
+      constexpr size_t bits=sizeof(size_t)*CHAR_BIT;
       size_t result=p.size();
-      for(unsigned i=0; i<p.size(); i++)
-        result^=p[i]<<i;
+      for(size_t i=0; i<std::min(p.size(), bits); i++)
+        result^=((size_t) p[i])<<i;
```

Comment:
I am also not quite sure whether the hashing works as intended. In particular I do not understand the left shift using the index in the `packedt` as parameter. For any `p.size() > 32/64` there are ignored values.

How about simply
```
result^=p[i];
```
which would at least use all values of `p`, or using a better hash geared towards hashing a collection of integral values?